patroni/postgresql/bootstrap.py

```diff
@@ -48,7 +48,7 @@ def option_is_allowed(name):
                     if len(keys) != 1 or not isinstance(opt[keys[0]], str) or not option_is_allowed(keys[0]):
                         error_handler('Error when parsing {0} key-value option {1}: only one key-value is allowed'
                                       ' and value should be a string'.format(tool, opt[keys[0]]))
-                    user_options.append('--{0}={1}'.format(keys[0], opt[keys[0]]))
+                    user_options.append('--{0}={1}'.format(keys[0], shell_quote(opt[keys[0]])))
```

Comment:
Last time I touched this method was at least 4 years ago and already forgot some details and @hughcapet probably never seen it yet.
It would be nice to add a docstring and type hints to process_user_options method, it will help the review.